features/step_definitions/basic_steps.rb

```diff
@@ -26,4 +26,12 @@
     within(dom_section) do
         click_on element
     end
+end
+
+Given("I fill in {string} field with {string}") do |element, value|
+    fill_in element,with: value
+end
+
+Given("I select {string} from Category") do |option|
+    select option, from: 'article[category_id]'
```

Comment:
This step could be made very generic if you use the `label` to choose what selector should be used. However, in your case, you are using `:category_id` on the select tag, rather than `category` (which leads to a mismatch) and Capybara can't make the right connection between the label and the selector. I'm not sure why you do it this way. It complicates things for you in testing but also in your view and your controller. Consider a refactor.

Reply:
refactors new and feature file to make this test universal